src/native/libs/System.Security.Cryptography.Native/pal_evp_kdf.c

```diff
@@ -0,0 +1,160 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#include "openssl.h"
+#include "pal_evp_kdf.h"
+#include "pal_utilities.h"
+
+#include <assert.h>
+
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wmissing-noreturn"
+void CryptoNative_EvpKdfFree(EVP_KDF* kdf)
+{
+#ifdef NEED_OPENSSL_3_0
+    if (API_EXISTS(EVP_KDF_free))
+    {
+        // No error queue impact
+        EVP_KDF_free(kdf);
+        return;
+    }
+#else
+    (void)kdf;
+#endif
+
+    assert(0 && "Inconsistent EVP_KDF API availability.");
+}
+#pragma clang diagnostic pop
+
+
+EVP_KDF* CryptoNative_EvpKdfFetch(const char* algorithm, int32_t* haveFeature)
+{
+    assert(haveFeature);
+
+#ifdef NEED_OPENSSL_3_0
+    if (API_EXISTS(EVP_KDF_fetch))
+    {
+        ERR_clear_error();
+        EVP_KDF* kdf = EVP_KDF_fetch(NULL, algorithm, NULL);
+
+        if (kdf)
+        {
+            *haveFeature = 1;
+            return kdf;
+        }
+        else
+        {
+            unsigned long error = ERR_peek_error();
+
+            // If the fetch failed because the algorithm is unsupported, then set
+            // haveFeature to 0. Otherwise, assume the algorithm exists and the
+            // fetch failed for another reason, and set haveFeature to 1.
+            *haveFeature = ERR_GET_REASON(error) == ERR_R_UNSUPPORTED ? 0 : 1;
+            return NULL;
+        }
+    }
+#else
+    (void)algorithm;
+    (void)haveFeature;
+#endif
+
+    *haveFeature = 0;
+    return NULL;
+}
+
+int32_t CryptoNative_KbkdfHmacOneShot(
+    EVP_KDF* kdf,
+    unsigned char* key,
+    int32_t keyLength,
+    char* algorithm,
+    unsigned char* label,
+    int32_t labelLength,
+    unsigned char* context,
+    int32_t contextLength,
+    unsigned char* destination,
```

Comment:
I believe we usually use `uint8_t` instead of `unsigned char` in our function signatures